web/src/app/webview/src/contextManager.ts

```diff
@@ -1,17 +1,15 @@
-import { JSKeyboard, Keyboard, OutputTargetInterface, Transcription, TextTransform } from 'keyman/engine/keyboard';
-// TODO-web-core: remove usage of OutputTargetBase, use OutputTargetInterface instead
-import { Mock, findCommonSubstringEndIndex, isEmptyTransform, OutputTargetBase } from 'keyman/engine/js-processor';
+import { JSKeyboard, Keyboard, TextStore, Transcription, TextTransform, SyntheticTextStore, findCommonSubstringEndIndex } from 'keyman/engine/keyboard';
 import { KeyboardStub } from 'keyman/engine/keyboard-storage';
 import { ContextManagerBase } from 'keyman/engine/main';
 import { WebviewConfiguration } from './configuration.js';
 import { LexicalModelTypes } from '@keymanapp/common-types';
-import { KMWString } from '@keymanapp/web-utils';
+import { KMWString, isEmptyTransform } from '@keymanapp/web-utils';
 
 export type OnInsertTextFunc = (deleteLeft: number, text: string, deleteRight: number) => void;
 
-export class ContextHost extends Mock {
+export class ContextHost extends SyntheticTextStore {
```

Comment:
Surprise!! `ContextHost` is also an `OutputTarget`!

I don't like the name `ContextHost`. What does it mean? It's a subclass of `Mock` but I don't understand why it exists. Can we find a different name or at least document it?

```suggestion
/**
 * huh? what is this class for?
 */
export class ContextHostTextStore extends SyntheticTextStore {
```

Reply:
`ContextHost` there is app/webview-specific - it has extra extensions explicitly designed to facilitate communication and synchronization with the host app, despite not being backed by any sort of Web element.